vuln-reach/src/javascript/project/mod.rs

```diff
@@ -548,6 +568,28 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn test_topo_sort_babel() {
+        let babel_register = npm_package("babel-register", "6.26.0").await;
+        let babel_core = npm_package("babel-core", "6.26.3").await;
+
+        let project = Project::new(
+            PackageResolver::builder()
+                .with_package("babel-register", babel_register)
+                .with_package("babel-core", babel_core)
+                .build(),
+            vec!["babel-core".to_string(), "babel-register".to_string()],
+        );
+
+        let vuln_node = VulnerableNode::new("babel-core", "lib/util.js", 72, 11, 72, 14);
+
+        // Reachability should error out because there is no topological ordering
+        // between the two packages with the current implementation, as they depend
+        // on each other. This doesn't mean that there is a runtime dependency cycle,
+        // but the respective `package.json` do point to each other.
+        assert!(project.reachability(&vuln_node).is_err());
```

Comment:
This is a valid project configuration. Why are we writing a test to confirm our parser is incorrect?

Reply:
This is a way of documenting the current state of the topological sort. This doesn't happen at the source level, but at the package dependency level, so the parser is not involved.

I am planning to change both this test and the behavior of the topological sort algorithm, to return an enum which will state whether cycles have been found.